File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.py
"""
return the kth largest element of the array
so it seems we cannot sort and then check because otherwise we would lose it

i know this is a heap problem so if I look it as a heap

I can heapify the array, then.............

i can heapop k times?
"""
import heapq
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        max_heap = []
        for n in nums:
            heapq.heappush(max_heap, -n)
        for i in range(k):
            res = -heapq.heappop(max_heap)
        return res

File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.rs
/*
kth largest element

so we cannot sort because the orer changes and we need the kth

but if we make a max heap and pop we get it
*/

use std::collections::BinaryHeap;
impl Solution {
    pub fn find_kth_largest(nums: Vec<i32>, k: i32) -> i32 {
        let mut heap: BinaryHeap<i32> = BinaryHeap::from(nums);
        for i in 0..k {
            let val = heap.pop();
            match val {
                Some(v)=>{
                    if i as i32 == k - 1 {
                        return v;
                    }
                }
                _ =>{}
            }
        }
        return 0;
    }
}